### ISSUE TITLE
How to find original path url before middleware append lang prefix to it?

### ISSUE THREAD

**Issue by nikola1970** [isaachinman/next-i18next#856]
Hello, I am trying to get the original requested path before middleware kicks in and append lang to it.

Want to somehow prevent 302 redirection if the page is 404. Old site was indexed with pages which do not have lang prefix appended and due to 302 redirect of the middleware those indexed pages wont be removed from the old google index because anything user type in the url bar will be redirected (302) to the default language. So `/some-not-found-page` will not respond with 404 but with 302 and then after redirection it will be 404.

So basically I need to know whether I typed `/some-route` or `/en/some-route` and hit enter.

Inspecting `req` object there is `originalUrl` which always gives back url with appended lang prefix and there is also `url` prop which will always return root of the path, no matter if I entered it with or without lang prefix. :(

Is there any solution for this? 

Thanks a lot!

**Comment by isaachinman** [comment]
I suppose you'd need to wrap your `appWithTranslation(MyApp)` in another HOC, and deal with it before the `next-i18next` code is called.

That being said, you might want to take a look at the link below, as i18n routing in NextJs will be changing very soon.

https://github.com/vercel/next.js/discussions/17078

**Comment by nikola1970** [comment]
Wrapping `appWithTranslation(MyApp)` in another HOC won't help since it all happens more low level, in the `server.js` file where I add this line ` server.use(nextI18NextMiddleware(nextI18next));`, and I can't find a way to save original entered url in the custom middleware before it and them pass it to `_app` `getInitialProps`. :(

**Comment by isaachinman** [comment]
The latest versions of this package do not use a middleware or custom servers.
